### PR TITLE
Add unit tests for com.twitter.graphjet.algorithms

### PR DESCRIPTION
**Pull request by paulkennethkent**
Hi,

I've analysed your codebase and seen some gaps in the coverage of the `com.twitter.graphjet.algorithms` module

I've written these tests for this module with the help of  [Diffblue](https://www.diffblue.com/) [Cover](https://static1.squarespace.com/static/5bc9dd6fb2cf795fa3511f70/t/5be2e84f8985836d62e8eb13/1541597264010/Diffblue+Cover+Datasheet.pdf).

Hopefully, these tests should help you detect regressions caused by future code changes.

### CODE REVIEW

**Review by JohnLBergqvist — APPROVED** [review]

File: graphjet-core/src/test/java/com/twitter/graphjet/algorithms/counting/tweetfeature/TweetFeatureTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+      // Act
+      final TweetFeature retval = TweetFeature.at(index);
+
+      // Assert result
+      Assert.assertEquals(TweetFeature.TWEET_FEATURE_SIZE, retval);
+    }
+
+    @Test
+    public void atInputZeroOutputNotNull000c81bcd3553cb198c() {
```
Comment (sudo-jaa) [inline graphjet-core/src/test/java/com/twitter/graphjet/algorithms/counting/tweetfeature/TweetFeatureTest.java]: We have a platform (test-gen?) ID in the test name here.

File: graphjet-core/src/test/java/com/twitter/graphjet/algorithms/filter/TweetTimeRangeFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import com.twitter.graphjet.stats.Counter;
+import com.twitter.graphjet.stats.StatsReceiver;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class TweetTimeRangeFilterTest {
+
+  @Test
+  public void originalTimeStampFromTweetIdInputPositiveOutputPositive00024a543da90d87bc5() {
```
Comment (sudo-jaa) [inline graphjet-core/src/test/java/com/twitter/graphjet/algorithms/filter/TweetTimeRangeFilterTest.java]: As before, we have another ID in the test name. I've been noticing this a lot on the test I have generated recently too. Is it a known bug?

File: graphjet-core/src/test/java/com/twitter/graphjet/algorithms/counting/tweetfeature/TweetFeatureTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package com.twitter.graphjet.algorithms.counting.tweetfeature;
+
+import com.twitter.graphjet.algorithms.counting.tweetfeature.TweetFeature;
+import org.junit.Assert;
+import org.junit.Rule;
```
Comment (sudo-jaa) [inline graphjet-core/src/test/java/com/twitter/graphjet/algorithms/counting/tweetfeature/TweetFeatureTest.java]: I believe junit.Rule for this class isn't necessary as we're not using timeout or expectedException

File: graphjet-core/src/test/java/com/twitter/graphjet/algorithms/filter/TweetTimeRangeFilterTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package com.twitter.graphjet.algorithms.filters;
+
+import com.twitter.graphjet.algorithms.filters.TweetTimeRangeFilter;
+import com.twitter.graphjet.stats.Counter;
+import com.twitter.graphjet.stats.StatsReceiver;
+import org.junit.Assert;
+import org.junit.Rule;
```
Comment (sudo-jaa) [inline graphjet-core/src/test/java/com/twitter/graphjet/algorithms/filter/TweetTimeRangeFilterTest.java]: I believe junit.Rule for this class isn't necessary as we're not using timeout or expectedException

File: graphjet-core/src/test/java/com/twitter/graphjet/algorithms/intersection/JaccardUpdateNormalizationTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package com.twitter.graphjet.algorithms.intersection;
+
+import com.twitter.graphjet.algorithms.intersection.JaccardUpdateNormalization;
+import org.junit.Assert;
+import org.junit.Rule;
```
Comment (sudo-jaa) [inline graphjet-core/src/test/java/com/twitter/graphjet/algorithms/intersection/JaccardUpdateNormalizationTest.java]: I believe junit.Rule for this class isn't necessary as we're not using timeout or expectedException

**Review by sudo-jaa — CHANGES_REQUESTED** [review]
Removal of junit.Rule import and IDs in test names. Otherwise these look awesome!

File: graphjet-core/src/test/java/com/twitter/graphjet/algorithms/filter/TweetTimeRangeFilterTest.java (hunk 2 above)
Comment (petersear) [reply graphjet-core/src/test/java/com/twitter/graphjet/algorithms/filter/TweetTimeRangeFilterTest.java]: I've seen this too. The following bugs have been logged:
https://diffblue.atlassian.net/browse/TG-4995
https://diffblue.atlassian.net/browse/TG-5497

Comment (sudo-jaa) [reply graphjet-core/src/test/java/com/twitter/graphjet/algorithms/filter/TweetTimeRangeFilterTest.java]: Looks good, I'll avoid logging any extras if it's already been recorded.

**Review by sudo-jaa — APPROVED** [review]

**Review by petersear — APPROVED** [review]

**Review by JohnLBergqvist — APPROVED** [review]

**Pull request closed without merging**